Sources/SwiftProtobuf/NameMap.swift

```diff
@@ -71,6 +71,57 @@ private class InternPool {
     }
 }
 
+/// Instructions used in bytecode streams that define proto name mappings.
+///
+/// This has package visibility so that it is also visible to the generator.
+package enum ProtoNameInstruction: UInt64 {
+    /// The proto (text format) name and the JSON name are the same string.
+    ///
+    /// ## Operands
+    /// * An integer representing the field or enum case number.
+    /// * A string containing the single text format and JSON name.
+    case same = 1
+
+    /// The JSON name can be computed from the proto string.
+    ///
+    /// ## Operands
+    /// * An integer representing the field or enum case number.
+    /// * A string containing the single text format name, from which the JSON name will be
+    ///   dynamically computed.
+    case standard = 2
+
+    /// The JSON and text format names are just different.
+    ///
+    /// ## Operands
+    /// * An integer representing the field or enum case number.
+    /// * A string containing the text format name.
+    /// * A string containing the JSON name.
+    case unique = 3
+
+    /// Used for enum cases only to represent a value's primary proto name (the first defined case)
+    /// and its aliases. The JSON and text format names for enums are always the same.
+    ///
+    /// ## Operands
+    /// * An integer representing the enum case number.
+    /// * An integer `aliasCount` representing the number of aliases.
+    /// * A string containing the text format/JSON name (the first defined case with this number).
+    /// * `aliasCount` strings containing other text format/JSON names that are aliases.
+    case alias = 4
+
+    /// Represents a reserved name in a proto message.
+    ///
+    /// ## Operands
+    /// * The name of a reserved field.
+    case reservedName = 5
+
+    /// Represents a range of reserved field numbers in a proto message.
+    ///
+    /// ## Operands
+    /// * An integer representing the lower bound (inclusive) of the reserved field number range.
+    /// * An integer representing the upper bound (exclusive) of the reserved field number range.
+    case reservedFields = 6
```

Comment:
"reservedNumber" nameMap isn't specific to fields, and enums also have reserved numbers, we just haven't needed to capture them.

Reply:
Renamed to `reservedNumbers`.